scripts/config/msu/classes/enum.nut

```diff
@@ -0,0 +1,49 @@
+::MSU.Class.Enum <- class
+{
+	_Keys = null;
+	_ValueKeyArray = null;
+
+	constructor( _array = null )
+	{
+		this._Keys = {};
+		this._ValueKeyArray = [];
+		if (_array != null) this.addArray(_array);
+	}
+
+	function _isValidKey( _key )
+	{
+		return typeof _key == "string" && _key != "" && _key.slice(0,1).tolower() != _key.slice(0,1);
+	}
+
+	function add( _key )
+	{
+		if (!this._isValidKey(_key))
+		{
+			::logError("MSU Enum Keys must be non-empty strings and start with a capitalized letter");
+			throw ::MSU.Exception.InvalidValue(_key);
+		}
+		this._Keys[_key] <- this._Keys.len();
+		this._ValueKeyArray.push(_key);
+	}
+
+	function addArray( _array )
+	{
+		foreach (value in _array) this.add(value);
+	}
+
+	function getKeyForValue( _value )
+	{
+		return this._ValueKeyArray[_value];
+	}
+
+	function len()
+	{
+		return this._Keys.len();
+	}
+
+	function _get( _key )
+	{
+		if (_key in this._Keys) return this._Keys[_key];
+		throw null;
```

Comment:
Why not throw InvalidKey?

Reply:
Sorry, forgot to point this out to you when you missed this, but basically https://github.com/MSUTeam/MSU/commit/50528f25488a5d93c5f2c8eec1f6c971275877fa. It means that squirrel will try and first check if there are built-in functions that match `_key` before throwing an error. It's in the doc for _get (2.15.2), so idk how I missed this before